### PR TITLE
Allow client script to post form data

### PR DESCRIPTION
**Pull request by aaannz**
this is useful when trying to upload some binary data like assets using
client script

### CODE REVIEW

**Comment by coolo** [comment]
I'm torn. your example sounds like misuse of the API :)

File: script/client [hunk 1]
```diff
@@ -138,11 +147,32 @@ else {
...
+    for (keys %params) {
+        if (/(\S+)\.(\S+)/) {
+            $form{$1}{$2} = $params{$_};
+        }
+        else {
+            $form{$_} = $params{$_};
+        }
+    }
+    %params = %form;
+}
```
Comment (okurz) [inline script/client]: That gota lot more complex. Can this be simplified?

Comment (coolo) [inline script/client]: huh? what do you want to simplify? And what is 'gota'? https://en.wikipedia.org/wiki/Gota gives me too many options

File: script/client [hunk 2]
```diff
@@ -138,11 +147,32 @@ else {
 }
 
 $url->path($path);
-$url->query([%params]) if %params;
+
+if (!$options{'form'}) {
```
Comment (coolo) [inline script/client]: just because perlcritic doesn't check scripts doesn't mean you can use quotes in hashes :)

File: script/client (hunk 1 above)
Comment (okurz) [inline script/client]: I wanted to write "got a" but somehow my android auto-corrected and I was too lazy to correct. I know my comment is not really helpful but to me it looks like many indendation levels now. Better just disregard my comment

**Comment by aaannz** [comment]
Indeed it is a misuse of API, you even need to patch job controller for it to work, so not a great example. I'll come with something else. I used this for stress testing openqa asset upload.

**Comment by aaannz** [comment]
Removed quotes from hash keys.
I couldn't come up with different example, we don't need nested form anywhere else.

**Comment by coolo** [comment]
perhaps leave this in unadvertised with a comment that this is useful for debugging?

**Comment by aaannz** [comment]
Done, commented it out and moved after POD data.

**Comment by coolo** [comment]
When you tidy: squash too :)

**Comment by aaannz** [comment]
Oh FFS :D tidy error in comment!
Done :)

**Pull request merged**